plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,277 @@
+
+
+#### Canonical Transaction
+
+A transaction is “canonical” if none of its inputs were previously spent in any other transaction. The definition of “previously spent” depends on whether or not the transaction in question is included in the Plasma chain.
+
+If the transaction was included in the chain, an input to that transaction would be considered previously spent if another transaction also spending the input was included in the chain *before* the transaction in question. Transaction position is determined by the tuple (block number, transaction index). If the transaction was not included in the chain, an input to that transaction would be considered previously spent if another transaction also spending the input is *known to exist*.
+
+Note that in this second case it’s unimportant whether or not the other transaction is included in the chain. If the other transaction is included in the chain, then the other transaction is clearly included before the transaction in question. If the other transaction is not included in the chain, then we can’t tell which transaction “came first” and therefore simply say that neither is canonical. 
+
+
+#### Exitable Transaction
+
+If a transaction is “exitable,” then a user may attempt to start an exit that references the transaction. All deposit transactions are “exitable” by default. A spend transaction can be called “exitable” if the transaction is correctly formed (e.g. more input value than output value, inputs older than outputs, proper structure) and is properly signed by the owners of the transaction’s inputs. 
+
+
+#### Valid Transaction
+
+A deposit transaction is “valid” if and only if it corresponds exactly to a single deposit on the root chain and is the first deposit transaction to correspond to that deposit. A spend transaction is “valid” if and only if it is exitable, canonical, and only stems from valid transactions (i.e. all transactions in the history are also valid transactions). Note that a transaction would therefore be considered invalid if even a single invalid transaction is present in its history. An exitable transaction is not necessarily a valid transaction.
+
+
+#### Correct Transaction
+
+A deposit transaction is “correct” if and only if it is valid. A spend transaction is “correct” if and only if it is exitable and only stems from valid transactions. All valid deposit transactions are also correct deposit transactions, but a spend transaction can be correct without also being valid in the case that the transaction is non-canonical. We make this distinction because it’s possible for a user to honestly sign a transaction that becomes non-canonical if another input to the transaction signs a second spend transaction. We later assert that users who only sign correct transactions are safe under our exit protocol.
+
+
+### Basic Specification
+
+The MoreVP exit protocol allows the owners of both inputs and outputs to a transaction to attempt an exit. 
```

Comment:
add: "...to ensure that the owner of the input can secure their funds in the case that the transaction results in being non-canonical (e.g., other inputs are invalid, outputs are invalid or users need to exit the Plasma chain before the transaction is processed)."